### ISSUE TITLE
gRing is declared as GRingPointType

### ISSUE THREAD

**Issue by mbjones** [NCEAS/eml#197]
---

Author Name: **Margaret O'Brien** (Margaret O'Brien)
Original Redmine Issue: 3227, https://projects.ecoinformatics.org/ecoinfo/issues/3227
Original Date: 2008-04-16
Original Assignee: Matt Jones

---

There was a short email exchange on the LTER best-practices mailing list about some confusion in the coverage schema, in the declaration for datasetGPolygonOuterRing. Peter and Wade may remember it. 

Basically, you have this:
<xs:element name="datasetGPolygonOuterGRing">
  <xs:complexType>
   <xs:choice>
     <xs:sequence>
       <xs:element name="gRingPoint" type="GRingPointType" minOccurs="3" maxOccurs="unbounded"/>
     </xs:sequence>
     <xs:element name="gRing" type="GRingPointType"/>
   </xs:choice>
  </xs:complexType> 
</element>

The sequence of 3 or more <gRingPoint>s (lat-lon pairs) creates a ring, but the odd thing is that the other choice, <gRing> is also declared as a gRingPointType. There is a named type GRingType (a list of ordered pairs) which appears not to have been used, so it seems that gRing should have been declared a gRingType instead. Wade suggested this, but it never made it into this database. 

If you all agree that this is indeed the case, I'll make the change and squeeze it into EML2.1.0rc3. It's is better located in 2.1.0, since it wont be backward compatible when it is fixed. Sorry for the timing - I ran into it today while diving back into an SBC project.

**Comment by mbjones** [comment]
---

Original Redmine Comment
Author Name: **Matt Jones** (Matt Jones)
Original Date: 2008-04-16T23:25:18Z

---

Yes, this is a bug and is basically unusable in its current state.  Changing the type definition as you suggest would allow either syntax to be used to delineate the gring.  Lets go ahead and fold it into 2.1 RC3 if you are so inclined.

**Comment by mbjones** [comment]
---

Original Redmine Comment
Author Name: **Margaret O'Brien** (Margaret O'Brien)
Original Date: 2008-04-17T17:37:33Z

---

*** Bug 3229 has been marked as a duplicate of this bug. ***

**Comment by mbjones** [comment]
---

Original Redmine Comment
Author Name: **Margaret O'Brien** (Margaret O'Brien)
Original Date: 2008-09-22T17:10:20Z

---

changing status to "fixed"

**Comment by mbjones** [comment]
---

Original Redmine Comment
Author Name: **Margaret O'Brien** (Margaret O'Brien)
Original Date: 2008-11-07T00:09:14Z

---

One other note:
The switch of GRing to GRingType was made months ago and this bug closed.
But I just noticed that GRing was a complexType that was completely empty, although the documentation states that it should have string content. It has been changed to simpleType in EML2.1. Probably this was never noticed in the past since this type was never used. I'm only putting a note here since it's related to the retyping of GRing.
